pkgs/development/libraries/science/math/amd-libflame/default.nix

```diff
@@ -0,0 +1,72 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gfortran
+, python3
+, amd-blis
+
+, withOpenMP ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "amd-libflame";
+  version = "2.2";
+
+  src = fetchFromGitHub {
+    owner = "amd";
+    repo = "libflame";
+    rev = version;
+    sha256 = "1s8zvq6p843jb52lrbxra7vv0wzmifs4j36z9bp7wf3xr20a0zi5";
+  };
+
+  patches = [
+    # The LAPACKE interface is compiled as a separate static library,
+    # we want the main dynamic library to provide LAPACKE symbols.
+    # This patch adds lapacke.a to the shared library as well.
+    ./add-lapacke.diff
+  ];
+
+  nativeBuildInputs = [ gfortran python3 ];
+
+  buildInputs = [ amd-blis ];
+
+  configureFlags = [
+    # Build a dynamic library with a LAPACK interface.
+    "--disable-static-build"
+    "--enable-dynamic-build"
+    "--enable-lapack2flame"
+
+    # Use C BLAS interface.
+    "--enable-cblas-interfaces"
+
+    # Avoid overloading maximum number of arguments.
+    "--enable-max-arg-list-hack"
+
+    # libflame by default leaves BLAS symbols unresolved and leaves it
+    # up to the application to explicitly link to a BLAS. This is
+    # problematic for us, since then the BLAS library becomes an
+    # implicit dependency. Moreover, since the point of the AMD forks
+    # is to optimized for recent AMD CPUs, link against AMD BLIS.
```

Comment:
Is there a way we can hook into @matthewbauer's nice `alternatives` implementation to provide `AMD BLIS` as a `blas` provider, the way we do for Intel MKL and openblas?

This gives you *one* overlay variable that you can update to slice *all* of NixPkgs consistently between openblas / mkl / etc.  The purpose here is to allow someone to slice a numpy / pandas / scipy / tensorflow / pytorch / etc. stack between MKL vs. openblas vs. whatever in O(1) package modifications instead of O(n) with the risk of missing some and ending in an incoherent build.

https://github.com/NixOS/nixpkgs/blob/21be830522ccf027bbf007ddf59409b2ecb1bfe9/doc/using/overlays.xml#L140-L262

Reply:
> Is there a way we can hook into @matthewbauer's nice `alternatives` implementation to provide `AMD BLIS` as a `blas` provider, the way we do for Intel MKL and openblas?

This should already work. E.g.:

```
❯ nix-build -E 'with import ./default.nix { overlays = [(self: super: { blas = (super.blas.override { blasProvider = self.amd-blis; }); lapack = (super.lapack.override { lapackProvider = self.amd-libflame; } ); } )]; }; finalfusion-utils'
/nix/store/3lljihln5hph2qxl4d5lhnpg0h53w05x-finalfusion-utils-0.11.2
❯ nix-store -qR result | egrep '(blis|flame)'
/nix/store/d9cssd4xp6d2jy5gazwfpksmpl1wc62v-amd-blis-2.2
/nix/store/66gx4lk596va3kvs97gbyyshyfhb42g8-amd-libflame-2.2
```

Reply:
That's fantastic!  Can you update the manual to advertise this nice feature?

Reply:
Will try to do that tomorrow!